README.md

```diff
@@ -0,0 +1,197 @@
+include proper CRC16 as data payload, and check received data CRC16 (or any
+other CRC).
+
+## Physical interface
+
+Serial interface is constructed from two DIO pins. Obviously one pin has to be
+configured for input (receiving), the other for output (transmitting). Usually
+NI-9401 module, providing total 8 DIOs, with split for 4 in and 4 output ports,
+is used.
+
+## LabView/Host interface
+
+[Common_FPGA_HealthAndStatus
+library](https://github.com/lsst-ts/Common_FPGA_HealthAndStatus/) is used for
+health and status FIFO. That provides datatype for the FIFO.
+
+For each use, one input and one output DIO are needed. Six FIFOs has to be
+provided - transmitting, receiving, timestamps, healt & status for Rx and Tx,
+and internal transmitting queue. Three boolean registers are needed - IRQ,
+WaitForIRQ, and ModbusTrigger (starting physical write after receving 0x8000
+"Wait for trigger command").
+
+FIFOs needs to be "never arbitrate" on side used inside module - write for
+outputs/receiving, read for inputs/transmitting. The other side should be set
+to never arbitrate as well, but that's isn't required.
+
+For each use:
+
+1. Create a U16 FIFO for received data.
+
+2. Create a U16 FIFO for transmitted data.
+
+3. Create a custom type ModbusTxInstruction FIFO for internal usage while
+   transmitting data. Specify never arbitrate for read and write.
+
+4. Create two custom type HealthAndStatusUpdate FIFOs - one for receiving,
+   second for transmission.
+
+5. Create a boolean register for IRQ control and "WaitForRXFrame". Specify
+   never arbitrate for write.
+
+6. Generate a boolean register for the software controlled trigger. Specify
+   never arbitrate for write. You can use a single trigger for controlling
+   multiple ports so this is an optional step if a register has already been
+   generated.
+
+7. Create a FIFO for Tx timestamps. Specify a data type of U64. Specify never
+   arbitrate for write.
+
+8. Drop the FPGAModbus/ModbusPort.vi onto your main VI. Setup the port settings
+   using the resources defined above. Sets base address (shall be multiply of
+   6) and IRQ (ordinals).  Review the comments in ModbusPort.vi for more
+   information.
+
+### Optional
+
+To copy health and status data FIFOs into a single FIFO:
+
+9. Drop the FPGAHealthAndStatus/HealthAndStatusFIFOCopy.vi. Set the internal
+   health and status FIFO. Set the source as the FIFO generated in step 2.
+
+10. Drop the FPGAHealthAndStatus/HealthAndStatusFIFOCopy.vi. Set the internal
+    health and status FIFO. Set the source as the FIFO generated in step 5.
+
+# Theory of operation
+
+FIFOs are used for what they are best suited - to provide a channel between
+variable speeds write and read ends. Usually one end of the FIFO is inside
+[SCTL](https://knowledge.ni.com/KnowledgeArticleDetails?id=kA00Z000000P8sWSAS&l=en-US).
+SCTL are used to grant execution of the commands inside FPGA in a single tick.
```

Comment:
Please consider "Usually one end of a FIFO is inside a [Single Cycle Timed Loop](...) (SCTL). SCTLs are used to grant"...